rust/datafusion/src/physical_plan/planner.rs

```diff
@@ -645,10 +653,11 @@ impl DefaultPhysicalPlanner {
                                         &list_expr_data_type,
                                         &value_expr_data_type,
                                     ) {
-                                        expressions::cast(
+                                        expressions::cast_with_options(
```

Comment:
Doesn't `expressions::cast` do the same thing? I wonder if we need to change this?